src/ResultSet.php

```diff
@@ -11,5 +11,5 @@
      *
      * @return int
      */
-    public function numFields(): int;
+    public function getNumFields(): int;
```

Comment:
Can this be added to `Amp\Sql\ResultSet` or does MySQL not provide this information upfront?

Reply:
Amp\Mysql has this: `public function getFields(): Promise`

But it's written with Amp\Socket, so we can probably refactor it.